packages/eslint-plugin/src/rules/strict-void-return.ts

```diff
@@ -0,0 +1,636 @@
+      nonVoidFuncInImplMember:
+        'Value-returning function provided as `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+      nonVoidFuncInOther:
+        'Value-returning function used in a context where a void function is expected.',
+      nonVoidFuncInReturn:
+        'Value-returning callback returned from a function, which must return a void callback.',
+      nonVoidFuncInVar:
+        'Value-returning function assigned to `{{varName}}`, which is declared as a void function.',
+
+      nonVoidReturnInArg:
+        'Value returned in a callback argument to `{{funcName}}`, which expects a void callback.',
+      nonVoidReturnInAttr:
+        'Value returned in `{{attrName}}` event handler prop of `{{elemName}}`, which expects a void event handler.',
+      nonVoidReturnInExtMember:
+        'Value returned in `{{memberName}}` method of `{{className}}`, whose base class `{{baseName}}` declares it as a void method.',
+      nonVoidReturnInImplMember:
+        'Value returned in `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+      nonVoidReturnInOther:
+        'Value returned in a context where a void return is expected.',
+      nonVoidReturnInReturn:
+        'Value returned in a callback returned from a function, which must return a void callback.',
+      nonVoidReturnInVar:
+        'Value returned in `{{varName}}`, which is declared as a void function.',
+    },
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          allowReturnAny: {
+            type: 'boolean',
+            description:
+              'Whether to allow functions returning `any` to be used in place expecting a `void` function.',
+          },
+          allowReturnNull: {
+            type: 'boolean',
+            description:
+              'Whether to allow functions returning `null` to be used in place expecting a `void` function.',
+          },
+          allowReturnPromiseIfTryCatch: {
+            type: 'boolean',
+            description:
+              'Whether to allow functions returning a promise if the function is na async function expression whose whole body is wrapped in a try-catch block. This offers an alternative to an async IIFE for handling errors in async callbacks.',
+          },
+          allowReturnUndefined: {
+            type: 'boolean',
+            description:
+              "Whether to allow functions returning `undefined` to be used in place expecting a `void` function. When disabled, `void` operator can't be used to discard the return value, because it evaluates to undefined. Disable this to enforce a consistent style across the codebase.",
+          },
+        },
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      allowReturnNull: false,
+      allowReturnPromiseIfTryCatch: true,
```

Comment:
[Feature] In this PR's OP, `allowReturnPromiseIfTryCatch` is only mentioned in the collapsed note around fixers & suggestions. Seems to me it's no longer necessary, right? 

ACK that it's not much added code but I don't think we have consensus that it's a strategy we'd want to add to the rule. Could you please split it out? We can always talk about it more in a followup issue - either its own and/or the one around fixers & suggestions.